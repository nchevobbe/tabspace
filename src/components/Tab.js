// @flow
import type { TabType, TabIdType } from "../types.flow";
import React from "react";

export default (props: {
  tab: TabType,
  onTabClick: TabIdType => Promise<TabType>,
  onCloseButtonClick: TabIdType => Promise<void>,
}) => {
  const { tab, onTabClick, onCloseButtonClick } = props;
  const { title, url, id, favIconUrl } = tab;
  return (
    <a
      className="tab"
      href={url}
      title={title}
      onClick={e => {
        e.preventDefault();
        onTabClick(id);
      }}
    >
      <div className="tab-thumbnail" />
      <img className="tab-favicon" src={favIconUrl} />
      <button
        className="tab-close clean"
        title="Close tab"
        onClick={e => {
          e.stopPropagation();
          onCloseButtonClick(id);
        }}
      />
      <span className="tab-title">{title}</span>
    </a>
  );
};
