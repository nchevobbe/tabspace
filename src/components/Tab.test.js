import React from "react";
import { shallow } from "enzyme";

import Tab from "./Tab";

describe("Tab", () => {
  it("renders", () => {
    expect(
      shallow(
        <Tab tab={{ title: "Title", url: "url", id: 1, favIconUrl: "path/to/img" }} />
      )
    ).toMatchSnapshot();
  });
});
