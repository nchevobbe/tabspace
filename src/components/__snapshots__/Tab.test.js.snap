// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    tab={
      Object {
        "favIconUrl": "path/to/img",
        "id": 1,
        "title": "Title",
        "url": "url",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tab-thumbnail"
        />,
        <img
          className="tab-favicon"
          src="path/to/img"
        />,
        <button
          className="tab-close clean"
          onClick={[Function]}
          title="Close tab"
        />,
        <span
          className="tab-title"
        >
          Title
        </span>,
      ],
      "className": "tab",
      "href": "url",
      "onClick": [Function],
      "title": "Title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "tab-thumbnail",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "tab-favicon",
          "src": "path/to/img",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "tab-close clean",
          "onClick": [Function],
          "title": "Close tab",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
          "className": "tab-title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "span",
      },
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tab-thumbnail"
          />,
          <img
            className="tab-favicon"
            src="path/to/img"
          />,
          <button
            className="tab-close clean"
            onClick={[Function]}
            title="Close tab"
          />,
          <span
            className="tab-title"
          >
            Title
          </span>,
        ],
        "className": "tab",
        "href": "url",
        "onClick": [Function],
        "title": "Title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "tab-thumbnail",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "tab-favicon",
            "src": "path/to/img",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "tab-close clean",
            "onClick": [Function],
            "title": "Close tab",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
            "className": "tab-title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "span",
        },
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
